server/lib/renderers/pdf.js

```diff
@@ -2,96 +2,160 @@
  * @description
  * This service is responsible for producing PDFs on the server, it accepts
  * handlebar templates and uses the html renderer to produce valid HTML - then
- * streaming this through the wkhtmltopdf application to produce PDFs.
+ * streaming this through the puppeteer application to produce PDFs.
  *
- * @requires wkhtmltopdf
+ * @requires puppeteer
  * @requires lodash
+ * @requires fs
  * @requires stream-to-promise
  * @requires path
  * @requires q
  * @requires process
  * @requires debug
  */
 
-const wkhtmltopdf = require('wkhtmltopdf');
+const pptr = require('puppeteer');
 const _ = require('lodash');
-const streamToPromise = require('stream-to-promise');
-const path = require('path');
-const process = require('process');
-const debug = require('debug')('renderer:pdf');
+const fs = require('fs');
+const debug = require('debug')('app');
 
 const html = require('./html');
+const resources = require('./resources');
 
 const headers = {
   'Content-Type' : 'application/pdf',
 };
 
-exports.render = renderPDF;
-exports.headers = headers;
-exports.extension = '.pdf';
+// load scripts and styles
+const { dbStyles, dbScripts } = environmentStyleAndScripts();
+
+// pptr options
+const pptrOptions = {
+  headless : true,
+  args : [
+    '--no-sandbox',
+    '--disable-web-security',
+    '--disable-gpu',
```

Comment:
Out of curiosity, do we actually need all these flags?  What happens if we remove them?